### PR TITLE
remove SRC for Baytrail and CherryTrail

### PR DESCRIPTION
**Pull request by plbossart**
remove SRC since it's just too big for the internal memories.
We keep the mixer w/ a regular PCM and a 'deep buffer' one (limited to 10ms periods due to memory sizes).

Tested on CHT w/ rt5640 and max98090 (Cyan Chromebook)

### CODE REVIEW

**Comment by plbossart** [comment]
Tentatively Fixes https://github.com/thesofproject/sof/issues/3063

File: tools/topology/sof-byt-codec.m4 [hunk 1]
```diff
@@ -54,10 +54,10 @@ DAI_ADD(sof/pipe-dai-playback.m4,
 
 # PCM Media Playback pipeline 3 on PCM 1 using max 2 channels of s32le.
 # 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-pcm-media.m4,
+PIPELINE_PCM_ADD(sof/pipe-low-latency-playback.m4,
```
Comment (lgirdwood) [inline tools/topology/sof-byt-codec.m4:57]: Ok, so "media" is remove here, but this also removes the mixer and SRC. Easiest fix is to create a "pipe-pcm-mixer" that is basically the same as "pipe-pcm-media" minus the SRC.

Comment (plbossart) [reply tools/topology/sof-byt-codec.m4:57]: I am really not happy with the topologies. The mixer should be the start of a new pipeline, not embedded in one specific pipeline. It's been 2 years and it's probably time to rethink all this...

**Comment by marc-hb** [comment]
> hence RFC status.

Someone showed me it's now possible to switch any PR back to "Draft" status. Github drafts:
- cannot be merged
- appear in a different color
- can be filtered out in queries with `draft:false` or `-is:draft` (forgot which one sorry)

I've been using github drafts since the feature was created and there's no drawback whatsoever. Let's use drafts.

Also:
- drafts do not spam CODEOWNERS automatically (you can still add anyone manually)

... so I always start every PR as a draft in case CI tells me I missed something, this saves a lot of email.

https://github.blog/2019-02-14-introducing-draft-pull-requests/
https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/

Off-topic sorry.

File: tools/topology/sof-byt-codec.m4 (hunk 1 above)
Comment (lgirdwood) [reply tools/topology/sof-byt-codec.m4:57]: Ignore my last comment, I thought mixer was in the media pipe, it's in the low latency pipe. Yep I agree, low-latency pipe needs renamed to include "mixer" and the PCM should be removed from it so it starts at the mixer.

**Comment by lgirdwood** [comment]
> This all looks good, but it looks like CI is attempting a hw params that are no longer supported ?

The same params are actually working on boards with a codec, but I dont see any immediate difference between the topologies.

**Comment by plbossart** [comment]
> > This all looks good, but it looks like CI is attempting a hw params that are no longer supported ?
> 
> The same params are actually working on boards with a codec, but I dont see any immediate difference between the topologies.

````
kernel: [  177.961849]  PCM: ASoC: no BE found for SSP2.OUT
kernel: [  177.962840]  PCM: ASoC: no backend DAIs enabled for PCM
````

Likely a mismatch between topology and machine driver. Will test, maybe I went too fast on the nocodec support

**Comment by plbossart** [comment]
> > > This all looks good, but it looks like CI is attempting a hw params that are no longer supported ?
> > 
> > 
> > The same params are actually working on boards with a codec, but I dont see any immediate difference between the topologies.
> 
> ```
> kernel: [  177.961849]  PCM: ASoC: no BE found for SSP2.OUT
> kernel: [  177.962840]  PCM: ASoC: no backend DAIs enabled for PCM
> ```
> 
> Likely a mismatch between topology and machine driver. Will test, maybe I went too fast on the nocodec support

What I thought, stupid edit mistake

````diff
 DAI_ADD(sof/pipe-mixer-dai-playback.m4,
-       1, SSP, SSP_NUM, SSP2-Codec,
+       1, SSP, SSP_NUM, NoCodec-2,
````

Fixed now, let's see how CI likes it.

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Comment by xiulipan** [comment]
Now the .TEXT size is 0x12ea7 on CHT, 0x12d27 on BYT

https://sof-ci.01.org/sofpr/PR3080/build6448/build/cht_xcc.txt
```
writing module 0 sof-cht

	Totals	Start		End		Size
	TEXT	0xff2c0000	0xff2d2ea7	0x12ea7
	DATA	0xff300010	0xff328000	0x27ff0
	BSS	0xff3023f0	0xff306258	0x3e68
```

https://sof-ci.01.org/sofpr/PR3080/build6448/build/byt_xcc.txt
```
writing module 0 sof-byt

	Totals	Start		End		Size
	TEXT	0xff2c0000	0xff2d2d27	0x12d27
	DATA	0xff300010	0xff328000	0x27ff0
	BSS	0xff302310	0xff306178	0x3e68
```

**Comment by lgirdwood** [comment]
Jenkins is BAT fails sound clean, could be startup DUT or Test equipment clock delta appearing as a higher frequency that settles quickly.
@zrombel any idea why CI would fail for ICL when this is unrelated.

**Comment by aiChaoSONG** [comment]
@lgirdwood Can we get this merged? The ICL error obviously has nothing to do with this PR.

**Comment by lgirdwood** [comment]
@aiChaoSONG it was a CI server issue, it's been rerun and is now good..

**Comment by marc-hb** [comment]
Note there's a new, unrelated regression on these  same platforms now: #3146

**Comment by marc-hb** [comment]
I think the wrong `_defconfig` files  have been used for the last 3- months: https://github.com/thesofproject/sof/pull/2711#issuecomment-656945319

**Comment by marc-hb** [comment]
PR #3167 just got merged and it turns **off** `CONFIG_OPTIMIZE_FOR_SIZE` again for baytrail (and maybe for others)

**Comment by plbossart** [comment]
> PR #3167 just got merged and it turns **off** `CONFIG_OPTIMIZE_FOR_SIZE` again for baytrail (and maybe for others)

Are you saying PR #3167 is good or bad? Can't follow your point

**Comment by marc-hb** [comment]
> Are you saying PR #3167 is good or bad?

I'm saying neither; I'm merely trying to mitigate any future .TEXT size confusion.

**Pull request merged**